c7n/resources/directory.py

```diff
@@ -30,6 +34,31 @@ class resource_type(object):
         filter_name = 'DirectoryIds'
         filter_type = 'list'
 
+    def augment(self, directories):
+
+        def _add_tags(d):
+            client = local_session(self.session_factory).client('ds')
+            try:
+                tags = client.list_tags_for_resource(
+                    ResourceId=d['DirectoryId']).get('Tags', [])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'AccessDenied':
```

Comment:
this error handling doesn't make any sense, if there's no resource policy on a type of resource, then a denied is a persistent error wrt to the custodian execution role permissions, that needs to bubble up. the error handling through the rest of the pr also seems suspect.

Reply:
ah, yes. you are correct.